sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/VectorIndexTest.java

```diff
@@ -0,0 +1,338 @@
+package com.azure.cosmos.rx;
+
+import com.azure.cosmos.ConsistencyLevel;
+import com.azure.cosmos.CosmosAsyncClient;
+import com.azure.cosmos.CosmosAsyncContainer;
+import com.azure.cosmos.CosmosAsyncDatabase;
+import com.azure.cosmos.CosmosClientBuilder;
+import com.azure.cosmos.CosmosDatabaseForTest;
+import com.azure.cosmos.CosmosException;
+import com.azure.cosmos.DirectConnectionConfig;
+import com.azure.cosmos.implementation.TestConfigurations;
+import com.azure.cosmos.implementation.guava25.collect.ImmutableList;
+import com.azure.cosmos.models.CosmosContainerProperties;
+import com.azure.cosmos.models.DistanceFunction;
+import com.azure.cosmos.models.Embedding;
+import com.azure.cosmos.models.ExcludedPath;
+import com.azure.cosmos.models.IncludedPath;
+import com.azure.cosmos.models.IndexingMode;
+import com.azure.cosmos.models.IndexingPolicy;
+import com.azure.cosmos.models.PartitionKeyDefinition;
+import com.azure.cosmos.models.VectorDataType;
+import com.azure.cosmos.models.VectorEmbeddingPolicy;
+import com.azure.cosmos.models.VectorIndexSpec;
+import com.azure.cosmos.models.VectorIndexType;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.annotations.AfterClass;
```

Comment:
Please also add tests doing rountrip serialization - like Policy -> json -> policy -> json and comaring that the strings are identical or similar. We found several issues in json serialization with this type of tests